services/requester/requester.go

```diff
@@ -152,6 +153,54 @@ func (e *EVM) SendRawTransaction(ctx context.Context, data []byte) (common.Hash,
 		return common.Hash{}, err
 	}
 
+	txDataLen := len(tx.Data())
```

Comment:
let's extract this logic into a seperate validate function, this way the function doesn't get too long and it's easy to test, that being said, let's add a unit test using this function to test for all the cases.

Reply:
Good point :+1: 
Updated in https://github.com/onflow/flow-evm-gateway/pull/314/commits/a851720798cbdaec7775d78c45941df070d25134